docs/DeveloperGuide.md

```diff
@@ -273,72 +273,99 @@ _{Explain here how the data archiving feature will be implemented}_
+| `* * * *` | Counsellor       | search for a patient                      | quickly access and review patient status.                   |
+| `* * * *` | Counsellor       | list patient at the given index           | quickly access patients that come regularly.                |
+| `* * *`   | Counsellor       | to categorise / tag my patients           | patients with more serious issues can be attended to first. |
+| `* * *`   | New User         | have a help function                      | so that I know how to use the application.                  |
+| `* *`     | Counsellor       | sort patients based on their priority tag | more serious patients can be attended first.                |
+| `*`       | Experienced User | mass delete patient data                  | patient data is not compromised.                            |
 
-*{More to be added}*
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `CogniCare` application and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: Search a student**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests to list persons using the specified constraints
+2.  CogniCare shows a list of persons that meets the criteria
 
     Use case ends.
 
 **Extensions**
 
+* 1a. The query has missing parameters
+
+    * 1a1. CogniCare shows an error message.
+
+    * 1a2. CogniCare returns all information about a patient.
+    
+    Use case ends.
+
 * 2a. The list is empty.
 
   Use case ends.
 
-* 3a. The given index is invalid.
 
-    * 3a1. AddressBook shows an error message.
+**Use case: Delete a student**
+
+**MSS**
+
+1.  User requests to delete a patient at the given index.
+2.  CogniCare displays the patient information prior to deletion, and confirms with the user prior to deletion.
+3.  CogniCare deletes the patient.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The query has missing parameters
+
+    * 1a1. CogniCare shows an error message that the index is invalid.
+
+    Use case ends.
+
+* 2a. User does not want to delete.
+    * 1a1. CogniCare shows an error message that the user cancelled the delete operation.
+  
+  Use case ends.
 
-      Use case resumes at step 2.
 
-*{More to be added}*
 
 ### Non-Functional Requirements
 
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
 2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
 
+
```

Comment:
```suggestion
```

Reply:
Thanks Khang Hou, sorry missed this comment. Will keep in mind for future.